repositories.md

```diff
@@ -5,10 +5,67 @@ css_id: repositories
+
+#### Changes to a Primary
+
+If the Time Server is implemented, the primary SHALL use the following procedure to verify the time. This procedure will occur after the vehicle version manifest is sent and will fulfill the "Download and check current time" step of the Uptane Standard.
+
+1. Gather the tokens from each secondary ECU's version report.
+2. Send the list of tokens to the Time Server to fetch the current time. The time server responds as described in [Time Server](#time_server), providing a cryptographic attestation of the last known time.
+3. If the Time Server's response meets the criteria below, update the primary ECU's clock and retain the Time Server's response for distribution to secondary ECUs, otherwise discard it and proceed without an updated time.  The criteria for checking the Time Server's response are:
+  - The signature over the Time Server's response is valid.
+  - The tokens provided to the Time Server have been included in the response.
+  - The time in the Time Server's response is later than the last time verified in this manner.
+
+#### ECU Version Report
+
+The payload of the ECU version report should contain the latest time downloaded from the Time Server. In addition, the report should include a token (nonce) for the Time Server to sign and send back.
+
+#### Changes to all ECUs
+
+At build time, ECUs will be provisioned with an attestation of the current time downloaded from the Time Server.
+
+As the first step to verifying metadata, described as "Load and verify the current time or the most recent securely attested time" in the Standard, the ECU SHOULD load and verify the  most recent time from the Time Server using the following procedure:
+
+1. Verify that the signatures on the downloaded time are valid.
+2. Verify that the list of nonces/tokens in the downloaded time includes the token that the ECU sent in its previous version report.
+3. Verify that the time downloaded is greater than the previous time.
+
+If all three steps complete without error, the ECU SHALL overwrite its current attested time with the time it has just downloaded, and generate a new nonce/token for the next request to the Time Server.
+
+If any check fails, the ECU SHALL NOT overwrite its current attested time, and SHALL jump to the last step ([Create and Send Version Report](https://uptane.github.io/uptane-standard/uptane-standard.html#create_version_report)), and report the error. The ECU MUST reuse its previous token for the next request to the Time Server.
+
+#### Changes to checking Root metadata
+
+After updating to the most recent Root metadata file, check the Timeserver key. If the Timeserver key is listed in the Root metadata and has been rotated, reset the clock used to determine the expiration of metadata to a minimal value (e.g. zero, or any time that is guaranteed to not be in the future based on other evidence).  It will be updated in the next cycle.
```

Comment:
Explain why.

Reply:
This was in a previous version of the standard, so I'm not sure why it was added. I think it is to ensure that the new timeserver does not accidentally cause a warning about a rollback.

Reply:
Yes, I meant please add this explanation to the doc :)